R/utils.R

```diff
@@ -68,6 +69,20 @@ plotRefMapToList <- function(map, plotRef) {
   dataType <- veupathUtils::toStringOrNull(map$dataType[map$plotRef == plotRef])
   dataShape <- veupathUtils::toStringOrNull(map$dataShape[map$plotRef == plotRef])
   displayLabel <- veupathUtils::toStringOrNull(map$displayLabel[map$plotRef == plotRef])
+  naToZero <- veupathUtils::toStringOrNull(map$naToZero[map$plotRef == plotRef])
+  
+  # Validate naToZero and fix if necessary
+  if (length(naToZero) == 0) {
+    warning("Encountered empty naToZero value. Setting naToZero = FALSE.")
+    naToZero <- FALSE
+  } else if (is.na(naToZero) || naToZero=='') {
+    warning("Encountered '' or NA as the naToZero value. Setting naToZero = FALSE.")
+    naToZero <- FALSE
+  } else {
+    if (identical(naToZero, 'TRUE')) naToZero <- TRUE
+    if (identical(naToZero, 'FALSE')) naToZero <- FALSE
+  }
+  if (!identical(naToZero, '') & !(naToZero %in% c(TRUE, FALSE))) {stop("plotRefMapToList error: Unrecognized value submitted for naToZero")}
```

Comment:
at this point shouldnt naToZero always be T/F ? whats the check for empty string here doing?